src/content/docs/en/guides/on-demand-rendering.mdx

```diff
@@ -232,6 +232,28 @@ if (!product) {
 </html>
 ```
 
+#### Redirect to another page
+
+Instead of returning a `Response` object, you can also redirect the user to another page using `Astro.redirect`:
+
+```astro title="src/pages/product/[id].astro" {10}
+---
+export const prerender = false; // Not needed in 'server' mode
+
+import { getProduct } from '../api';
+
+const product = await getProduct(Astro.params.id);
+
+// No product found
+if (!product) {
+  return Astro.redirect('/404', 404);
+}
+---
+<html>
+  <!-- Page here... -->
+</html>
+```
+
```

Comment:
I'm not sure about this suggestion, that's why I made a separate commit. Since we present the different possible `Response` formats with on demand rendering, I thought that `Astro.redirect` also has its place here. Especially since the second parameter is only applicable with on demand rendering. On the other hand, this is perhaps not the best example... I took the one above.

Reply:
Doesn't returning a 404 response automatically navigate them to the 404 page anyway?

I'm not 100% sure here

Reply:
Yes I think so too... that's why I said it might not be the best example. I did think of a 301, but I was looking for an example without making it too complicated; it didn't come to me at the time. I'll think about it! (well probably tomorrow, it's getting late for me ^^)

Reply:
A quick idea that came to me: (I also added a link to `Astro.redirect` reference)
````suggestion
#### Redirect to another page

Instead of returning a `Response` object, you can also redirect the user to another page using [Astro.redirect](/en/reference/api-reference/#astroredirect):

```astro title="src/pages/product/[id].astro" {10}
---
export const prerender = false; // Not needed in 'server' mode

import { UNRENEWED_PRODUCTS } from '../constants';
import { getProduct } from '../api';

const product = await getProduct(Astro.params.id);

// The product no longer exists
if (UNRENEWED_PRODUCTS.includes(product)) {
  return Astro.redirect('/products', 301);
}
---
<html>
  <!-- Page here... -->
</html>
```

````

Reply:
What about just modifying the existing example slightly?

```ts
---
export const prerender = false; // Not needed in 'server' mode

import { getProduct } from '../api';

const product = await getProduct(Astro.params.id);

// No product found
if (!product) {
  Astro.response.status = 404;
  Astro.response.statusText = 'Not found';
  // Or redirect the user to another page
  return Astro.redirect('/products', 301);
}
---
<html>
  <!-- Page here... -->
</html>
```

Reply:
I think you're right, there's no need to add a new section! However, it seems to me that we always propose valid examples so we need another `if` for the code to be valid. I'm going to post a new commit with a new proposal for the ``Return a `Response` object`` section.